sharedUiComponents/lines/textInputLineComponent.tsx

```diff
@@ -64,14 +64,18 @@ export class TextInputLineComponent extends React.Component<ITextInputLineCompon
 
     updateValue(value: string) {
         if (this.props.numbersOnly) {
-            if (/[^0-9\.\p\x\%]/g.test(value)) {
+            let checkValue = value;
+            if (value.substr(0, 1) === "-") {

```

Comment:
```suggestion
            if (value[0] === "-") {
```

Avoid creating a new string (and then recreating another one one line after :-))